docs/reference-guide/modules/events/pages/event-handlers.adoc

```diff
@@ -14,89 +14,395 @@ If the event handler does not need access to the payload of the message, you can
 Do not configure the payload type on the annotation if you want the payload to be passed as a parameter.
 For a complete list of all parameters, we refer to xref:messaging-concepts:supported-parameters-annotated-handlers.adoc#supported-parameters-for-event-handlers[this] section.
 
-In all circumstances, at most one event handler method is invoked per listener instance.
-Axon will search for the most specific method to invoke, using the following rules:
+[[handler-resolution]]
+== Handler resolution
 
-. On the actual instance level of the class hierarchy (as returned by `this.getClass()`), all annotated methods are evaluated.
-. If the framework finds one or more methods for which it can resolve all parameters to a value, the method with the most specific type is chosen and invoked.
-. If the framework finds no methods on this level of the class hierarchy, the supertype is evaluated the same way.
-. When it reaches the top level of the hierarchy and no suitable event handler is found, the event is ignored.
+When an event is published, Axon determines which event handler methods to invoke based on the event type.
+In Axon 5, **all matching event handlers are invoked** for each event.
 
-Consider the following Event Handling Component sample to explain the event handler invocation rules further:
+The framework uses the following rules to determine matching handlers:
+
+. Axon identifies all event handlers that can handle the event type (based on the event's `MessageType`).
+. For each matching handler, Axon evaluates whether it can resolve all required parameters.
+. All handlers for which parameters can be resolved are invoked.
+. If no suitable event handler is found, the event is ignored (or handled by the processor's default behavior).
+
+NOTE: This is a change from Axon 4, where only the most specific handler method was invoked per instance.
+In Axon 5, all matching handlers are invoked, allowing multiple handlers in the same component to process the same event.
+
+Consider the following Event Handling Component sample:
 
 [source,java]
 ----
-// assume EventB extends EventA 
-// and    EventC extends EventB
-// and    a single instance of SubListener is registered
+public class OrderEventHandler {
 
-public class TopListener {
+    @EventHandler
+    public void on(OrderPlacedEvent event) {
+        // Update read model
+    }
 
     @EventHandler
-    public void handle(EventA event) {
+    public void logOrderPlaced(OrderPlacedEvent event, @MetadataValue("userId") String userId) {
+        // Log the event - this handler is ALSO invoked for OrderPlacedEvent
     }
 
     @EventHandler
-    public void handle(EventC event) {
+    public void on(OrderShippedEvent event) {
+        // Handle order shipped
     }
+}
+----
+
+In this example, when an `OrderPlacedEvent` is published, **both** the `on()` and `logOrderPlaced()` methods will be invoked.
+
+[[message-types-and-event-names]]
+== Understanding message types and event names
```

Comment:
Providing a description on how to interact with the `MessageType` of an event makes sense, but this level of detail is a bit much. We have other sections already explaining this (for example, the "Anatomy of a Message" chapter). It's fine to have a couple of sentence and a small sample, but for more detailed options, we should revert to that section instead.